dashboard/models.py

```diff
@@ -150,14 +151,45 @@ def get_supported_courses(self):
             logger.info("Courses did not exist", exc_info = True)
         return []
 
+    def earliest_start_datetime(self) -> Optional[datetime.datetime]:
+        """Get the earliest start date of courses in the QuerySet
 
-class CourseManager(models.Manager):
-    def get_queryset(self):
-        return CourseQuerySet(self.model, using=self._db)
+        :return: Earliest start date of courses in the QuerySet, or None if no course dates found
+        :rtype: datetime
+        """
+        sorted_courses = sorted(self.all(), key=lambda course: course.course_date_range.start)
 
-    def get_supported_courses(self):
-        return self.get_queryset().get_supported_courses()
+        earliest_start = None
+        if len(sorted_courses) > 0:
+            earliest_course = sorted_courses[0]
+            earliest_start = earliest_course.course_date_range.start
+            logger.info(f"Earliest start datetime for all courses: {earliest_start.isoformat()} found in course {earliest_course.canvas_id}")
+        else:
+            logger.info(f"No course listed. Returning None as the earliest_start_datetime_of_course. ")
+        return earliest_start
 
+    def update_all_last_cron_run(self):
+        """ Updates the last cron run to now for all courses in QuerySet
+        """
+        # Finally update the time on all courses updated
+        self.update(last_cron_run=datetime.datetime.now())
```

Comment:
This is throwing a warning. I think we should use `datetime.datetime.now(tz=pytz.UTC)` or `datetime.datetime.now(tz=timezone.utc)` (with the latter, you need `from django.utils import timezone`).

```
RuntimeWarning: DateTimeField Course.last_cron_run received a naive datetime (2020-09-10 16:11:00.695774) while time zone support is active.
```

Reply:
Should we be using a timestamp from the beginning of the cron run? Feels like this way we might lose some data if the events occur during the run?

Reply:
Yeah, good point it should probably lock down the time at the beginning. If there's a little duplicate next run it's not a problem but if it's too late then it is. I'll capture the time at the start of the run.